### ISSUE TITLE
\p should not have any trailing spaces before the newline

### ISSUE THREAD

**Issue by birchamp** [unfoldingWord/translationCore#7164]
Currently USFM exports text incorrectly. It puts a space or two after '\p' If this marker is on a line by itself then it should be followed by a newline immediately. Note: \p may be followed by text on the same line, but typically it is just by itself.

**Comment by da1nerd** [comment]
@birchamp does this look like what you want?

```diff
\v 4 In this way they may train the younger women to love their own husbands and children,
+ \p
- \p  
\v 5 From Abraham to King David, the following ancestors are listed: Abraham,
Isaac, Jacob, Judah and his brothers; then Perez and Zerah (their mother was Tamar*),
Hezron, Ram, Amminadab, Nahshon, Salmon, Boaz (his mother was Rahab*), Obed (his
mother was \jmp Ruth|link-href="#article-Ruth"\jmp*), Jesse, and King David.
\p some inline text
```

**Comment by birchamp** [comment]
@da1nerd yes, that looks right to me, but I'd like @RobH123  to take a look at it to be sure.

**Comment by RobH123** [comment]
Oh yes, thanks @da1nerd. I guess when you think you're mostly done, I'd like a quick look thru some whole file outputs (e.g., email them or put them in a branch or something) before you move on to other things.

**Comment by elsylambert** [comment]
@mannycolon Can you please help me locate where the build is to test 3.0.2 issues?

**Comment by mannycolon** [comment]
I'm not sure where build 3.0.2 can be found. It looks like both 3.0.2 and 3.1.0 got merged into the same develop branch before a release branch was created for 3.0.2. I would ask @birchamp @da1nerd @PhotoNomad0 any further questions

**Comment by elsylambert** [comment]
@birchamp I tested this issue in translationCore 3.0.2 (e2d8c61). I imported a project from D43 online(which has white space in its usfm), but after the online import, the new project's usfm does not have white space after /p and in  |x .
<img width="863" alt="Screen Shot 2021-11-05 at 1 42 47 PM" src="https://user-images.githubusercontent.com/51806180/140555024-2bd31511-33a2-4d29-918f-8f0ba6214be9.png">
 But this works only for online import, not for local import. Neither does it work for Export to usfm. When exported to usfm, it is still adding white spaces
<img width="976" alt="Screen Shot 2021-11-05 at 1 44 13 PM" src="https://user-images.githubusercontent.com/51806180/140555271-20fbbc65-431a-49f9-8b90-921647917fe7.png">
.

**Comment by elsylambert** [comment]
@da1nerd does this issue cover only online imports?

**Comment by da1nerd** [comment]
I think we only got as far as discussing and identifying the changes to be made. This hasn't been implemented yet.

**Comment by PhotoNomad0** [comment]
@da1nerd It looks like https://github.com/unfoldingWord/usfm-js/pull/101 is the fix for this?

**Comment by PhotoNomad0** [comment]
@da1nerd It looks like https://github.com/unfoldingWord/usfm-js/pull/101 is the fix for this?

**Comment by da1nerd** [comment]
wow I guess' it's been quite a while. I totally forgot about that PR.

**Comment by PhotoNomad0** [comment]
@elsylambert verified in translationCore 3.0.2 (c8c2941)

**Comment by elsylambert** [comment]
Verified in translationCore 3.0.2 (c8c2941). \p still adds white spaces after Export to USFM. These are two usfm files that I had exported. The projects(source usfm) does not have any spaces after \p.
[67-REV-test2.usfm.zip](https://github.com/unfoldingWord/translationCore/files/7570797/67-REV-test2.usfm.zip)

[43-LUK-test.usfm.zip](https://github.com/unfoldingWord/translationCore/files/7570796/43-LUK-test.usfm.zip)

**Comment by PhotoNomad0** [comment]
@elsylambert are these usfm files the source or the output?  if not the source, could you zip up the projects so I could reproduce the problem?  Thanks.

**Comment by elsylambert** [comment]
@PhotoNomad0 ^^ those were output files after the export to USFM. Here are the zipped source projects:
[en_new_rev_book.zip](https://github.com/unfoldingWord/translationCore/files/7570972/en_new_rev_book.zip)
[en_loc_luk_book.zip](https://github.com/unfoldingWord/translationCore/files/7570973/en_loc_luk_book.zip)

**Comment by PhotoNomad0** [comment]
@elsylambert Some good test projects here.  Actually the projects do have a space after the \p marker.  Will have to find a way to fix this in the USFM export since we do not want this.

<img width="626" alt="Screen Shot 2021-11-19 at 4 12 28 PM" src="https://user-images.githubusercontent.com/14238574/142693602-6288a7c0-8aef-4a07-b125-b1c879fc55c9.png">

**Comment by PhotoNomad0** [comment]
@elsylambert Also FYI - do not pay attention to the USFM files in the base project folder.  The two in red are old orphan files either copies of original USFM import or generated by previous DCS uploads before project was renamed.  The one in orange was created the last time you uploaded to DCS and will only be uploaded on DCS uploads.  The recent data used by tCore are stored in json files in the rev folder.

<img width="415" alt="Screen Shot 2021-11-19 at 4 21 59 PM" src="https://user-images.githubusercontent.com/14238574/142694100-9b460a7c-17b6-45f5-8252-7af2ca27b428.png">

**Comment by PhotoNomad0** [comment]
@elsylambert - Have a fix for the projects you found in `translationCore 3.0.2 (0d9cd25)`.

**Comment by elsylambert** [comment]
Looks good in translationCore 3.0.2 (0d9cd25). Not seeing trailing spaces.

**Comment by gayathrihayden** [comment]
Fix tested in tC 3.0.2 (0d9cd25)  and works fine. 

Reference screenshots:

In tC 3.0.1 (343198b), notice trailing space after \p:

![tC_3_0_1-trailing_space p](https://user-images.githubusercontent.com/74929193/143137878-85746d5d-cebd-417c-8c33-46ee0b30a3db.PNG)

In tC 3.0.2 (0d9cd25), do not observe trailing space after \p:

![tC_3_0_2-no trailing_space character p](https://user-images.githubusercontent.com/74929193/143137637-4e6f64f1-3b56-4af2-aa81-e0f290d50f0f.PNG)
